### ISSUE TITLE
Make Loading accepts parameter

### ISSUE THREAD

**Issue by jihchi** [lrosa007/remotedata-re#1]
I'm thinking about make `Loading` type accepts parameter:

```reason
type t('a, 'p, 'e) =
  | NotAsked
  | Loading('p)
  | Failure('e)
  | Success('a);
```

For example, this is for "load more", "infinite scroll" or "pagination" cases.

When `state` already fetched partial data, its type should be `Success('a)` and renders data listing.
If I "load more" data, I've to change type to `Loading` which currently didn't hold parameter,
screen will renders spinner or placeholder which UX is bad.
I would like to renders spinner also previous data listing.

Did you think that make sense?

**Comment by rvcas** [comment]
@jihchi yea that sounds cool. PR is welcome ❤️ 

This was at first more of a learning project but if this is lib is useful you, I'll be more active on it. I'm also happy to add you as a collaborator and give you push access.

**Comment by jihchi** [comment]
@lrosa007 Thank you! I'll create a PR.

**Comment by rvcas** [comment]
@jihchi cool sounds great

**Comment by rvcas** [comment]
closed by #2
